STIPS/STIP-005.md

```diff
@@ -0,0 +1,877 @@
+   </td>
+   <td valign="top">
+     <p> The redeemer brings only the SetToken. While we only return to the redeemer the USDC value of the amount of asset that he had exposure to. </p>
+     <p> There is an AMM involved in this process. </p>
+   </td>
+  </tr>
+  <tr>
+    <td valign="top">Frequency of rebalances</td>
+    <td valign="top">
+        <p> In CLM/ALM, since our position was a CDP, we had to actively manage our leverage ratios, keep track of them and call lever/delever every 24 hours. </p>
+   </td>
+   <td valign="top">
+    <p> Leveraged Perp positions are subject to funding flows which may move the Set substantially off its leverage target (negatively or positively). They're also exposed to liquidation risks when account values drop below minimum requirements. </p>
+    <p> The module will need to support active maintenance of the leverage target although it's not immediately clear what the rebalance cadence might be since this depends on actual market behavior. </p>
+   </td>
+  </tr>
+</table>
+
+### PerpV2 issuance and redemption flow analysis
+
+#### Issuance
+
+<table>
+  <tr>
+   <td width="20%"><strong>Issuance Step</strong>
+   </td>
+   <td><strong>Action</strong>
+   </td>
+  </tr>
+  <tr>
+   <td>Initial call flow
+   </td>
+   <td>
+<ul>
+
+<li>User calls <strong><em>issue(minQuantity) </em></strong>on DIM
+
+<li><em>DIM</em> calls <em>PerpV2Module</em> issuance hook
+
+<li><em>PerpV2Module</em> calls <strong><em>sync()</em></strong> inside the module issuance hook
+</li>
+</ul>
+   </td>
+  </tr>
+  <tr>
+   <td><em>sync() </em>
+<p>
+(flow summary)
+   </td>
+   <td>
+  <ul>
+  <li>Reads the virtual token balances of the virtual SetToken from PerpV2Module</li>
+  <li>Updates the default position units  on the virtual SetToken </li>
+    <ul>
+      <li>Only done if the new unit calculation does not equal existing unit </li>
+      <li>Only necessary if a liquidation has occurred. </li>
+      <li>Interest accrual is not an issue here </li>
+    </ul>
+  <li>Updates external USDC position on the real SetToken </li>
+  </ul>
+   </td>
+  </tr>
+<tr>
+   <td>sync()
+  <p>(updating virtual default positions if necessary)
+   </td>
+   <td><em>newUnitForVirtualAsset = vSetToken.balanceOf(vAsset) / realSetToken.totalSupply()</em>
+   </td>
+  </tr>
+  <tr>
+   <td>sync()
+  <p>(Update external USDC position on the real SetToken)
+   </td>
+   <td>
+     <p> First determine the additional amount of virtual assets we need exposure to: </p>
+     <p> &nbsp;&nbsp;<em> newVAssetUnit = mintQuantity * vSetToken.getDefaultPositionRealUnit(vAsset) </em> </p>
+     <p> Calculate the amount of USDC required to buy newVAssetUnit by simulating a trade on Perp’s vAMM. The delta quote returned by the simulation will include realized funding payments and fees. These will be distributed across the real SetToken's external USDC position. </p>
```

Comment:
Is there a function we can call to get the realized funding payments separately?

Reply:
Yes. We can call [Exchange.getPendingFundingPayment][1] to retrieve the payment that will be realized when the trade executes. 

And we can get an oracle-like price with [Lens/Quoter.swap][2] which lets us get the expected amount out or amount in without executing the swap. (This is what the Perp team recommended although they caution that the method is expensive to run)


[1]: https://github.com/perpetual-protocol/perp-lushan/blob/7c7974e2a830117528f08598899d2f6c00167f6e/contracts/Exchange.sol#L363-L379
[2]: https://github.com/perpetual-protocol/perp-lushan/blob/7c7974e2a830117528f08598899d2f6c00167f6e/contracts/lens/Quoter.sol#L56-L141